src/components/shared/PageHero.tsx

```diff
@@ -0,0 +1,122 @@
+import { css } from '@emotion/react';
+import styled from '@emotion/styled';
+
+import { FC } from 'react';
+import { Heading } from '~/components';
+import {
+  atMinLg,
+  atMinDesktop,
+  atMinLargeDesktop,
+  atMinTablet,
+  bp,
+  BreakpointName,
+} from '~/theme/breakpoints';
+import { spacing } from '~/theme/spacing';
+import { cssClamp } from '~/theme/util';
+
+type HeroImageSource = {
+  largeDesktop: string;
+  tablet: string;
+  xl: string;
+  xs: string;
+};
+
+type HeroImageProps = {
+  alt: string;
+  imgSource: HeroImageSource;
+};
+
+type PageHeroProps = HeroImageProps & {
+  text: string;
+};
+
+const Image = styled.picture`
+  margin-left: calc(50% - 50vw);
+
+  ${atMinLargeDesktop} {
+    margin-left: -10rem;
+  }
+
+  img {
+    filter: brightness(0.75);
+    height: ${cssClamp([17.5, 'smMobile'], [20.188, 'mobile'], [35, 'tablet'])};
+  }
+`;
+
+const HeaderText = styled(Heading)`
+  bottom: 4.5rem;
+  position: relative;
+
+  ${atMinTablet} {
+    bottom: 13.75rem;
+    font-size: 6.25rem;
+  }
+
+  ${atMinLg} {
+    max-width: 50rem;
+  }
+
+  ${atMinDesktop} {
+    bottom: 30rem;
+    left: 32rem;
+  }
+
+  ${atMinLargeDesktop} {
+    left: 42rem;
+  }
+`;
+
+const ImageContainer: FC<HeroImageProps> = ({ alt, imgSource }) => {
+  const sortedImgSourcesDescending = Object.entries(imgSource).sort(
+    ([breakpointA], [breakpointB]) => {
+      const valueA = breakpointA as BreakpointName;
+      const valueB = breakpointB as BreakpointName;
+      return bp[valueB] - bp[valueA];
```

Comment:
💬 I think that the name "key" rather than "value" makes more sense here.

```suggestion
      const keyA = breakpointA as BreakpointName;
      const keyB = breakpointB as BreakpointName;
      return bp[keyB] - bp[keyA];
```

Reply:
That makes way more sense, thanks!